fission/package.go

```diff
@@ -147,7 +152,21 @@ func updatePackage(client *client.Client, pkg *crd.Package, envName, envNamespac
 	if len(envName) > 0 {
 		pkg.Spec.Environment.Name = envName
 		pkg.Spec.Environment.Namespace = envNamespace
-		needToBuild = true
+
+		// we need to fetch the env to find out if it has a builder image, only then set needToBuild to true
```

Comment:
This is a separate problem, and it's bigger than this flag.  Simply avoiding a build doesn't really help the user here at all.  A package that has only source and now way to build it isn't useful.  Can you track a separate issue for this?  We should catch this case in function/package creation.

Reply:
yeah i agree it isn't really useful to not build. but this happens only on fn/pkg update with just `--env` flag. 
if the user has provided a `--src` flag, then the needToBuild gets set to true in `if len(srcArchive)` condition. 

but we cant catch it during creation right? this happens only on updates. but what we can do is do the GET operation on the env in fnUpdate and pkgUpdatem functions and send another param to this updatePackage function denoting if the env reference passed has a builder image or is just a deploy image.

what do you think?

Reply:
If it's not useful, let's not do it.  Let's figure out something that actually helps the user, and do that.

We could add some sort of validation error for functions that have source being associated with envs that have no builder.  This validation will have to be run during function and package creation, and during updates of functions, packages, and environments.

Reply:
ok, makes sense.